conference.js

```diff
@@ -1106,11 +1113,76 @@ export default {
     videoSwitchInProgress: false,
 
     /**
+     * This fields stores a handler which will create a Promise which turns off
+     * the screen sharing and restores the previous video state (was there
+     * any video, before switching to screen sharing ? was it muted ?).
+     *
+     * Once called this fields is cleared to <tt>null</tt>.
+     * {@type Promise|null}
+     */
+    _untoggleScreenSharing: null,
+
+    /**
+     * Creates a Promise which turns off the screen sharing and restores
+     * the previous state described by the arguments.
+     *
+     * This method is bound to the appropriate values, after switching to screen
+     * sharing and stored in {@link _untoggleScreenSharing}.
+     *
+     * @param {boolean} didHaveVideo indicates if there was a camera video being
+     * used, before switching to screen sharing.
+     * @param {boolean} wasVideoMuted indicates if the video was muted, before
+     * switching to screen sharing.
+     * @return {Promise} resolved after the screen sharing is turned off, or
+     * rejected with some error (no idea what kind of error, possible GUM error)
+     * in case it fails.
+     * @private
+     */
+    _turnScreenSharingOff(didHaveVideo, wasVideoMuted) {
+        this._untoggleScreenSharing = null;
+        this.videoSwitchInProgress = true;
+        return new Promise((resolve, reject) => {
+            APP.remoteControl.receiver.stop();
+            if (didHaveVideo) {
+                createLocalTracks({ devices: ['video'] })
+                    .then(([stream]) => this.useVideoStream(stream))
+                    .then(() => {
+                        JitsiMeetJS.analytics.sendEvent(
```

Comment:
I've made attempt to slightly simplify, but still not there yet. Can we start again from the current version of this method ?

The reason it was wrapped to set the in progress flag to false.

About the analytics where should they end up exactly ?